code/code/misc/disease.cc

```diff
@@ -1723,6 +1718,68 @@ int disease_pneumonia(TBeing *victim, int message, affectedData *af)
   return FALSE;
 }
 
+
+// extreme pain makes use of AFF_STUNNED for its practical effects
+//
+// applies&removes an AGI penalty manually, as the affect code can't handle adding
+// that kind of stuff to a disease affect
+int disease_extreme_pain(TBeing *vict, int stage, affectedData *aff)
+{
+  switch (stage) {
+
+    case DISEASE_BEGUN:
+      if (vict->task)
+        vict->stopTask();
+      if (vict->spelltask)
+        vict->stopCast(STOP_CAST_NONE);
+      if (vict->riding && IS_SET_DELETE(vict->fallOffMount(vict->riding, POSITION_STUNNED), DELETE_THIS))
+        return DELETE_THIS;
+      act("$n screams horribly and falls to the $g!", false, vict, nullptr, nullptr, TO_ROOM);
+      act("Agony overcomes you and you scream and fall to the $g!", false, vict, nullptr, nullptr, TO_CHAR);
+      act("$n is rendered senseless by the pain of $s wounds.", false, vict, nullptr, nullptr, TO_ROOM);
+      act("You are rendered senseless by the pain of your wounds.", false, vict, nullptr, nullptr, TO_CHAR);
+      vict->setPosition(POSITION_STUNNED);
+      vict->addToStat(STAT_CURRENT, STAT_AGI, -EXTREME_PAIN_AGI_PENALTY);
```

Comment:
I'm doing it the same way the stock affect code does it, just "by hand", so it will work as well as the rest of that does. I do wonder if your stats get reset to STAT_NATURAL on death or if the code just wipes all the affects off and lets that take care of it, but I don't know.